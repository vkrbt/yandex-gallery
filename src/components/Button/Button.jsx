import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export const Button = props => <button {...props} className={classnames('button', props.className)} />;

Button.propTypes = {
  className: PropTypes.string,
}
