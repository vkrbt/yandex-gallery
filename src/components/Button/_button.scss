.button {
  border: none;
  outline: none;
  background: transparent;
}
