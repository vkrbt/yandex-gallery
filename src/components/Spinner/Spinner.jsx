import React from 'react';

const Spinner = () => (
  <div className="spinner-wrapper">
    <div className="spinner" />
  </div>
);

export default Spinner;
