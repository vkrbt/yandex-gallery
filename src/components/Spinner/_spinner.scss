@keyframes spin {
  from {
    transform: rotateZ(0);
  }
  to {
    transform: rotateZ(360deg);
  }
}

.spinner-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;

  width: 100%;
  height: 200px;
}

.spinner {
  width: 40px;
  height: 40px;

  animation: spin 1s linear infinite;

  border: 2px solid #555;
  border-top-color: transparent;
  border-radius: 50%;
}
