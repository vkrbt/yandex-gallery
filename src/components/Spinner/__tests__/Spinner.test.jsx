import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from '../Spinner';
import { Button } from '../../Button/Button';

describe('Spinner', () => {
  it('should render spinner if loading', () => {
    const wrapper = shallow(<Spinner loading text="" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render button with text if not loading', () => {
    const text = 'Any text';
    const wrapper = shallow(<Spinner loading={false} text={text} />);
    expect(wrapper.find(Button)).toHaveLength(1);
    expect(wrapper.find(Button).first().childAt(0).text()).toBe(text);
  });
});
