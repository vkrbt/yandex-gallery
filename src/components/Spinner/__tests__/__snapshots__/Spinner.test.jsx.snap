// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render spinner if loading 1`] = `
<div
  className="loading-wrapper"
>
  <div
    className="spinner"
  />
</div>
`;
